crates/consensus/beacon/src/engine/error.rs

```diff
@@ -1,26 +1,45 @@
+use reth_miner::error::PayloadError as BuilderError;
 use reth_rpc_types::engine::PayloadError;
 use reth_stages::PipelineError;
 use thiserror::Error;
 
 /// Beacon engine result.
 pub type BeaconEngineResult<Ok> = Result<Ok, BeaconEngineError>;
 
+// TODO: add conversions to codes for engine spec compliance
+// one notable variant would be UnknownPayload
 /// The error wrapper for the beacon consensus engine.
 #[derive(Error, Debug)]
 pub enum BeaconEngineError {
     /// Forkchoice zero hash head received.
     #[error("Received zero hash as forkchoice head")]
     ForkchoiceEmptyHead,
+    /// Invalid payload attributes.
+    #[error("Invalid payload attributes")]
+    InvalidPayloadAttributes,
     /// Pipeline channel closed.
     #[error("Pipeline channel closed")]
     PipelineChannelClosed,
+    /// Unknown payload
+    #[error("Unknown payload")]
+    UnknownPayload,
     /// Encountered a payload error.
     #[error(transparent)]
     Payload(#[from] PayloadError),
+    /// Encountered an error during the payload building process.
+    #[error(transparent)]
+    BuilderError(#[from] BuilderError),
```

Comment:
naming is still not great,
I want to rename `reth_miner::error::PayloadError` perhaps `PayloadBuilderError`